assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -1885,6 +1887,9 @@ public SubmissionStatus getSubmissionCanonicalStatus(AssignmentSubmission submis
     private AssignmentConstants.SubmissionStatus getGradersCanonicalSubmissionStatus(AssignmentSubmission submission) {
         if (submission == null) return SubmissionStatus.NO_SUBMISSION;
 
+        String resubmissionString = StringUtils.trimToNull(submission.getProperties().get(AssignmentConstants.ALLOW_RESUBMIT_NUMBER));
+        boolean resubmissionAllowed = resubmissionString != null ? Integer.parseInt(resubmissionString) > 0 : false;
```

Comment:
parseInt() can throw NumberFormatException, this should probably be caught and logged.

Reply:
Hi @plukasew , I don't think this was totally necessary because the value of "submission.getProperties().get(AssignmentConstants.ALLOW_RESUBMIT_NUMBER)" should be always a number. I was even contemplated the possibility of this value could be null. I agree that is safer if we check it is really a number, but I don't think there is a chance to not a number being returned by this property.

Anyways, I have changed it to improve it, thanks!

Reply:
I have used "NumberUtils.isCreatable()" to ensure the string is a number.

Reply:
@plukasew There are literally 0 cases where this value could contain letters or float strings, the only way that this could happen is if the input inserted by the user is not properly validated. So, the good way to fix this (if there is really any chance that this could fail), it would be to validate the user input instead of validating it every time the value is used.

Reply:
@plukasew Can you prove that there is a chance that the stored value in here could be a float number or letter-containing? If you can prove it, I will change my code.